modules/gdscript/gdscript_parser.cpp

```diff
@@ -1089,6 +1089,26 @@ void GDScriptParser::parse_class_body(bool p_is_multiline) {
 				advance();
 				if (previous.get_identifier() == "export") {
 					push_error(R"(The "export" keyword was removed in Godot 4. Use an export annotation ("@export", "@export_range", etc.) instead.)");
+				} else if (previous.get_identifier() == "tool") {
+					push_error(R"(The "tool" keyword was removed in Godot 4. Use the "@tool" annotation instead.)");
+				} else if (previous.get_identifier() == "onready") {
+					push_error(R"(The "onready" keyword was removed in Godot 4. Use the "@onready" annotation instead.)");
+				} else if (previous.get_identifier() == "remote") {
+					push_error(R"(The "remote" keyword was removed in Godot 4. Use the "@rpc" annotation with the "mode" parameter set to "any_peer" instead.)");
+				} else if (previous.get_identifier() == "remotesync") {
+					push_error(R"(The "remotesync" keyword was removed in Godot 4. Use the "@rpc" annotation with the "mode" parameter set to "any_peer" and the "sync" parameter set to "call_local" instead.)");
+				} else if (previous.get_identifier() == "puppet") {
+					push_error(R"(The "puppet" keyword was removed in Godot 4. Use the "@rpc" annotation with the "mode" parameter set to "authority" instead.)");
+				} else if (previous.get_identifier() == "puppetsync") {
+					push_error(R"(The "puppetsync" keyword was removed in Godot 4. Use the "@rpc" annotation with the "mode" parameter set to "authority" and the "sync" parameter set to "call_local" instead.)");
+				} else if (previous.get_identifier() == "master") { // Not sure what the Godot 4 version would be?
+					push_error(R"(The "master" keyword was removed in Godot 4.)");
+				} else if (previous.get_identifier() == "mastersync") { // Not sure what the Godot 4 version would be?
+					push_error(R"(The "mastersync" keyword was removed in Godot 4.)");
+				} else if (previous.get_identifier() == "slave") { // Only in Godot 3.0 (deprecated in 3.1), equivalent to "puppet"?
+					push_error(R"(The "slave" keyword was removed in Godot 4. Use the "@rpc" annotation with the "mode" parameter set to "authority" instead.)");
+				} else if (previous.get_identifier() == "sync") { // Only in Godot 3.0?
+					push_error(R"(The "sync" keyword was removed in Godot 4. Use the "@rpc" annotation with the "sync" parameter set to "call_local" instead.)");
```

Comment:
`slave` and `sync` were deprecated ages ago (with `slave` being a controversial keyword), so I'd prefer not to bring them back in our codebase seemingly forever.

Reply:
Makes sense. I'll remove those shortly! 🙂